tests/functional/test_wps_package.py

```diff
@@ -1378,6 +1382,9 @@ def test_execute_application_package_process_with_bucket(self):
             "baseCommand": "echo",
             "arguments": ["$(runtime.outdir)"],
             "requirements": {
+                CWL_REQUIREMENT_APP_DOCKER: {
+                    "dockerPull": "alpine:latest"
+                },
```

Comment:
The tests flagged a legit problem. This conflicts with the below `CWL_REQUIREMENT_APP_BUILTIN` in hints.
It can only be one of the two. 
There should probably be something that explicitly blocks deployment of a "builtin" process (if not already the case?) when using the Web-API.

Otherwise, this is a hack around that works just during tests since some execution mocking is happening.

Reply:
In this case, the `hints` are supposed to _"ensure remote files are downloaded prior to CWL execution"_ but the [CWL documentation](https://www.commonwl.org/v1.0/CommandLineTool.html#Requirements_and_hints) explains that the `hints` report just a warning if an error occurs. 
Should we move the section hints to the requirements?

Reply:
The reason that it was placed in `hints` is exactly for this reason. If in `requirements`, `BuiltinRequirement` causes an error because it is unknown. 
The problem here is more about the Weaver API than CWL. It looks for `requirements`/`hints` and tries to match only 1 of the [constant process types](https://github.com/crim-ca/weaver/blob/7c23830eba030e86b12d538b38de6a49d53345b6/weaver/processes/constants.py#L20-L34), in order to figure out how to instantiate it with the corresponding class.